pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
making wait default to true for *some* resources is what seems off... kubectl just let the server do whatever it was going to do previously... on some resources (namespaces and pods) the server deleted asynchronously. on others, it was synchronous. this seems like trying to snapshot server behavior at a point in time and make it be enforced with resource-specific client-side logic.